src/es6/SirenAction.js

```diff
@@ -116,7 +116,7 @@ const _performSirenAction = function(action, fields, tokenValue) {
 	let body;
 
 	if (fields) {
-		fields = _appendHiddenFields(action, fields);
+		_appendHiddenFields(action, fields);
```

Comment:
Might it be better to remove all the field-setting logic from `_getEntityUrl`, so it just uses the fields passed in directly, and then move the `const url` assignment to below line 122?
If I read it correctly, at that point `fields` should contains all fields, both hidden and not, and avoids duplicating logic and work just to avoid some side-effects.